### DISCUSSION TITLE
Using TTS programmatically in python

### DISCUSSION THREAD

**Discussion by corlangerak** [coqui-ai/TTS#2123] [General Q&A]
This is probably a very noob question...

but how can I use TTS models programmatically within python code?
I cannot find it within the documentation, but I might be overlooking something.

**Comment by thorstenMueller** [comment]
Depending on ;-).
Would you like TTS fully integrated or is it okay to run a TTS server on your network and access this within your python code?

If it's the second maybe you can check out https://pypi.org/project/cTTS/

**Reply by corlangerak** [reply]
Thank you! This is certainly helpful, although how would I set up a server... sorry if this is a silly question.
Furthermore, I would really like to be able to fully integrate it within my code and have it run locally.
any ideas on where I could look?

**Reply by thorstenMueller** [reply]
I won't miss any chance to "promote" my voice tech Youtube channel 😆.
I made a video on how simple it is to setup a Coqui TTS server here: https://youtu.be/alpI-DnVlO0

**Comment by lexkoro** [comment]
You can take this as starting point https://github.com/coqui-ai/TTS/blob/dev/TTS/bin/synthesize.py
Especially maybe the [Synthesizer](https://github.com/coqui-ai/TTS/blob/dev/TTS/bin/synthesize.py#L309) and which parameter it needs.
